tee-worker/litentry/core/data-providers/src/graphql.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright 2020-2022 Litentry Technologies GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::sgx_reexport_prelude::*;
+
+use crate::{
+	base_url::{GRAPHQL_AUTH_KEY, GRAPHQL_URL},
+	build_client, Error, HttpError,
+};
+use http::header::{AUTHORIZATION, CONNECTION};
+use http_req::response::Headers;
+use itc_rest_client::{
+	http_client::{DefaultSend, HttpClient},
+	rest_client::RestClient,
+	RestGet, RestPath,
+};
+use serde::{Deserialize, Serialize};
+use std::{
+	default::Default,
+	format, str,
+	string::{String, ToString},
+	vec,
+	vec::Vec,
+};
+
+pub struct GraphQLClient {
+	client: RestClient<HttpClient<DefaultSend>>,
+}
+
+impl Default for GraphQLClient {
+	fn default() -> Self {
+		Self::new()
+	}
+}
+
+pub enum VerifiedCredentialsNetwork {
+	Litentry,
+	Litmus,
+	Polkadot,
+	Kusama,
+	Khala,
+	Ethereum,
+}
+pub struct VerifiedCredentialsIsHodlerIn {
+	pub addresses: Vec<String>,
+	pub from_date: String,
+	pub network: VerifiedCredentialsNetwork,
+	pub token_address: String,
+	pub mini_balance: f64,
```

Comment:
It's a bit confusing that the input is a vector of addresses but only with one single network.

There's an `address` in `IsHodlerOutStruct` too, what is that? 

I think it's helpful if we can attach some exemplary request/response as comments

Reply:
Sure. It's a bit confusing. Will add some example/comments.